gossip/src/duplicate_shred.rs

```diff
@@ -127,6 +132,24 @@ where
         return Ok(());
     }
 
+    // Chained merkle root conflict check
+    let first_shred = std::cmp::min_by(shred1, shred2, |s1, s2| {
+        s1.fec_set_index().cmp(&s2.fec_set_index())
+    });
+    let second_shred = std::cmp::max_by(shred1, shred2, |s1, s2| {
+        s1.fec_set_index().cmp(&s2.fec_set_index())
+    });
+
+    if let Some(erasure_meta) = ErasureMeta::from_coding_shred(first_shred) {
+        if allow_chained_duplicate_proofs
```

Comment:
we need to feature gate this because until chained merkle shreds are turned on, this check can be exploited to mark every block as duplicate.

Reply:
Shouldn't this whole thing be inside `if allow_chained_duplicate_proofs { ... }` block?

Reply:
also can you please include this github comment in the code as a comment?